tests/spec/fs.read.spec.js

```diff
@@ -88,3 +88,96 @@ describe('fs.read', function() {
     });
   });
 });
+
+describe('fs.promises.read', function() {
+  beforeEach(util.setup);
+  afterEach(util.cleanup);
+
+  it('should be a function', function() {
+    var fsPromises = util.fs().promises;
+    expect(fsPromises.read).to.be.a('function');
+  });
+
+  it('should return a promise', function() {
+    var fsPromises = util.fs().promises;
+    expect(fsPromises.read()).to.be.a('Promise');
+  });
+  it('should read data from a file', function() {
+    var fsPromises = util.fs().promises;
+    var wbuffer = new Filer.Buffer([1, 2, 3, 4, 5, 6, 7, 8]);
+    var rbuffer = new Filer.Buffer(wbuffer.length);
+    var fdesc;
+    rbuffer.fill(0);
+
+    return fsPromises.open('/myfile', 'w+')
+      .then((fd)=>{
+        fdesc=fd;
+        return fsPromises.write(fd, wbuffer, 0, wbuffer.length, 0);
+      })
+      .catch((error)=>{throw error;})
+      .then((result)=>{
+        expect(result).to.equal(wbuffer.length);
+        return fsPromises.read(fdesc, rbuffer,0,rbuffer.length,0);
+      })
+      .catch((error)=>{expect(error).not.to.exist;})
+      .then((result)=>{
+        expect(result).to.equal(rbuffer.length);
+        expect(wbuffer).to.deep.equal(rbuffer);
+      })
+    ;
+  });
+
+  it('should update the current file position', function() {
+    var fsPromises = util.fs().promises;
+    var wbuffer = new Filer.Buffer([1, 2, 3, 4, 5, 6, 7, 8]);
+    var rbuffer = new Filer.Buffer(wbuffer.length);
+    rbuffer.fill(0);
+    var _result = 0;
+    var fdesc;
+
+    return fsPromises.open('/myfile', 'w+')
+      .then((fd)=>{
+        fdesc=fd;
+        return fsPromises.write(fd, wbuffer, 0, wbuffer.length, 0 );
+      })
+      .then((result)=>{
+        expect(result).to.equal(wbuffer.length);
+        return fsPromises.read(fdesc, rbuffer, 0, rbuffer.length / 2, undefined);
+      })
+      .catch((error) => {throw error;})
+      .then((result)=>{
+        _result += result;
+        return fsPromises.read(fdesc,rbuffer,rbuffer.length / 2, rbuffer.length, undefined);
+      })
+      .catch((error)=>{expect(error).not.to.exist;})
+      .then((result)=>{
+        _result+=result;
+        expect(_result).to.equal(rbuffer.length);
+        expect(wbuffer).to.deep.equal(rbuffer);
+      })
+    ;
+  });
+
+  it('should fail to read a directory', function() {
+    var fsPromises = util.fs().promises;
+    var buf = new Filer.Buffer(20);
+    var buf2 = new Filer.Buffer(20);
+    buf.fill(0);
+    buf2.fill(0);
+
+    return fsPromises.mkdir('/mydir')
+      .then(()=>{return fsPromises.open('/mydir', 'r');})
+      .catch((error)=>{throw error;})
```

Comment:
A friendly suggestion: if you can remove this line because you don't need to catch error here.